File: aadf_tender_backend/src/main/java/com/team/models/DocumentVersion.java
package com.team.models;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name = "document_versions")
@Setter
@Getter
public class DocumentVersion extends BaseEntity {
    private String docType;
    private String url;
    private int version;
    private LocalDateTime uploadedAt;

    @ManyToOne
    @JoinColumn(name = "uploaded_by", nullable = false)
    private User uploadedBy;

    @ManyToOne
    @JoinColumn(name = "tender_id")
    private Tender tender;

    @ManyToOne
    @JoinColumn(name = "offer_id")
    private Offer offer;
}

File: aadf_tender_frontend/src/app/model/document-version.ts
import {Offer} from "./Offer";
import {Tender} from "./Tender";
import {User} from "./user";
import {BaseEntity} from "./base-entity";

export class DocumentVersion extends BaseEntity {
    docType: string;
    url: string;
    version: number;
    uploadedAt: Date;
    uploadedBy: User;
    tender?: Tender;
    offer?: Offer;
}
